api/get.js

```diff
@@ -0,0 +1,179 @@
+var _ = require('underscore'),
+    Q = require('q'),
+    glob = require('glob'),
+    fs = require('fs'),
+    path = require('path'),
+    request = require('request'),
+    config = require('config'),
+    
+    restify = require('restify'),
+    
+    overriders = require('./overriders'),
+    
+    folders = require('./folders'),
+    equals = require('./comparator');
+
+function Api(urlList) {
+    this.folder = folders.path;
+    this.urlList = urlList;
+}
+Api.prototype = {
+    handle: function(req, res, next) {
+        if (!this.urlList.hasOwnProperty(req.params.path)) {
+            res.send(404, 'Not handled!');
+            return next(new restify.errors.ResourceNotFoundError('Not Handled', '"' + req.params.path + '" is not defined by configuration'));
+        }
+        
+        var proxy = this.urlList[req.params.path];
+        overriders.save.folders(proxy, proxy.route);
+        this.folder = folders[proxy.name];
+        this.proxy(proxy, req, res, next)
+            .then(function() {
+                console.log('resolved', arguments);
+                return next();
+            })
+            .fail(function(error) {
+                console.error(error);
+                return next(error);
+            });
+    },
+    
+    
+    proxy: function(proxy, req, res, next) {
+        var url = proxy.url;
+        var patterns = [ path.join(folders[proxy.name], '*.req') ];
+        overriders.override.if.needed(patterns, proxy, this);
+        return Q.allSettled(_.map(patterns, function(pattern) {
+            return Q.nfcall(glob, pattern, { nosort: true, nodir: true });
+        }, this))
+            .then(function(fileList) {
+                fileList = _.chain(fileList).map(function(result) { return result.value; }).flatten().value();
+                this.find(fileList, req, res, next)
+                    .then(function(file) {
+                        console.log('then', path.basename(file), 'found!');
+                        return Q.promise(function(resolve) { resolve(file); });
+                    })
+                    .fail(function() {
+                        console.log('fail', 'NOT found!');
+                        return this.request(url, fileList, req, res, next);
+                    }.bind(this));
+            }.bind(this));
+    },
+    
+    find: function(fileList, req, res, next) {
+        var promises = _.map(fileList, function(file) {
+            return Q.Promise(function(resolve, reject) {
+                fs.readFile(file, { encoding: 'utf8' }, function(err, data) {
+                    if (err) {
+                        reject(err);
+                        return;
+                    }
+                    equals(req.params, data)
```

Comment:
The result of equals() isn't used, which looks suspicious. Is it modifying the stated of the passed args?